partials/KafkaConfig.java

```diff
@@ -0,0 +1,108 @@
+{% macro kafkaConfig(asyncapi) %}
+{%- set hasSubscribe = false -%}
+{%- set hasPublish = false -%}
+{%- for channelName, channel in asyncapi.channels() -%}
+    {%- if channel.hasPublish() -%}
+        {%- set hasPublish = true -%}
+    {%- endif -%}
+    {%- if channel.hasSubscribe() -%}
+        {%- set hasSubscribe = true -%}
+    {%- endif -%}
+{%- endfor -%}
+package com.asyncapi.infrastructure;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.serialization.IntegerSerializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.annotation.EnableKafka;
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory;
+import org.springframework.kafka.config.KafkaListenerContainerFactory;
+import org.springframework.kafka.core.*;
+import org.springframework.kafka.listener.ConcurrentMessageListenerContainer;
+import org.springframework.kafka.support.serializer.JsonDeserializer;
+import org.springframework.kafka.support.serializer.JsonSerializer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Configuration
+{% if hasPublish %}@EnableKafka{% endif %}
+public class Config {
+{%- if hasSubscribe or hasPublish %}
+    @Value("${kafka.bootstrap-servers:localhost:9092}")
+    private String bootstrapServers;
+{% endif %}
+{%- if hasPublish %}
+    @Value("${kafka.subscribe.pool-timeout:3000}")
+    private long poolTimeout;
+
+    @Value("${kafka.subscribe.amount-of-listeners:3}")
+    private Integer amountOfListeners;
```

Comment:
I don't see how this way of configuring Kafka will scale. Kafka has a lot of properties, is it not better keep the default spring-kafka way of provisioning properties?

Reply:
Took from this [docs of spring-kafka](https://docs.spring.io/spring-kafka/reference/html/#record-listener). This template is not cast in granite and this PR is just a first step which will allow biggest contribution, at least since of update to support spec 2.0